spec.bs

```diff
@@ -2941,17 +3005,29 @@ of the following global objects:
     1. If |result| is a [=ECMAScript/normal completion=]:
       1. Let |generatedBidIDL| be the result of [=converted to an IDL value|converting=]
         |result|'s \[[Value]] to a {{GenerateBidOutput}}.
-      1. If no exception was [=exception/thrown=] in the previous step, then set |generatedBid| to
-        the result of [=converting GenerateBidOutput to generated bid=] with |generatedBidIDL|,
-        |ig|, |expectedCurrency|, |isComponentAuction|, and |global|'s
-        [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=].
-      1. Otherwise, set |generatedBid| to failure.
+      1. If no exception was [=exception/thrown=] in the previous step:

```

Comment:
(to helper reviewer understand why making this change):
if generateBid()'s returned bid has some problem, we use the last setBid()'s valid bid if there's one, instead of simply setting the final bid to failure.